gcsfs/credentials.py

```diff
@@ -209,20 +213,6 @@ def connect(self, method=None):
             None,
         ]:
             self._connect_token(method)
-        elif method is None:
```

Comment:
Was there a reason to move this block to GCSFileSystem? Something in the tests seems to be getting here with `None` and failing.

Reply:
yes, the reason i initially moved this outside the `GoogleCredentials` class was that that chunk of code tires to call the `self.ls` method, which is only implemented for the `GCSFileSystem` class. now that we've deprecated the `self.check_credentials` argument, we technically move it back into the `GoogleCredentials` class, even though I personally prefer having the credential guessing outside the credentials class.

about the test failure: it should be enough the delete the `test_googlecredentials_none` test case or to make it assert that an error is raised. providing `token=None` is not valid anymore, since the credential guessing happens outside of this class.